File: Sorting and Searching/ResturantCustomer.cpp
#include<bits/stdc++.h>
    using namespace std;

int main() {
  int n;
  cin >> n;
  vector<int> arrival(n);
  vector<int> dept(n);
  for (int i = 0; i < n; i++) 
    cin >> arrival[i] >> dept[i];
  sort(arrival.begin(),arrival.end());
  sort(dept.begin(),dept.end());
  int i=1;
  int j=0;
  int count=1;
  int max=-99;
  while(i<n && j<n)
    {
        if(arrival[i]<=dept[j]){
            count++;
            i++;
        }
        else if (arrival[i] > dept[j]) { 
            count--; 
            j++; 
        }
            
        if(count>max)
            max=count;

    }
  cout << max << endl;
}

File: Sorting and Searching/ResturantCustomer.java
import java.io.*;
import java.util.*;
public class ResturantCustomer{
    public static void main(String args[])throws IOException
    {
    InputStreamReader r=new InputStreamReader(System.in);    
    Scanner sc=new Scanner(System.in);
    int n=sc.nextInt();
    long arrival[]=new long[n];
    long dept[]=new long[n];
    for(int i=0;i<n;i++){
        arrival[i]=sc.nextLong();
        dept[i]=sc.nextLong();
    }
    Arrays.sort(arrival);
    Arrays.sort(dept);
    long count=1;
    long max=Integer.MIN_VALUE;
    int j=0;
    int i=1;
    while(i<n && j<n)
    {
        if(arrival[i]<=dept[j]){
            count++;
            i++;
        }
        else if (arrival[i] > dept[j]) { 
            count--; 
            j++; 
        }
            
        if(count>max)
            max=count;

    }
    System.out.println(max);
}
}
